Sources/AWSXRayInstrument/Span.swift

```diff
@@ -0,0 +1,72 @@
+import AWSXRayRecorder
+import Baggage
+import Dispatch
+import Instrumentation
+
+extension XRayRecorder.Segment: Instrumentation.Span {
+    public var operationName: String {
+        // TODO: impl
+        "test"
+    }
+
+    public var kind: SpanKind {
+        // TODO: impl
+        .internal
+    }
+
+    public var status: SpanStatus? {
+        // TODO: impl
+        get {
+            nil
+        }
+        set(newValue) {}
+    }
+
+    public var startTimestamp: DispatchTime {
+        // TODO: impl
+        DispatchTime.now()
+    }
+
+    public var endTimestamp: DispatchTime? {
+        // TODO: impl
+        nil
+    }
+
+    public func end(at timestamp: DispatchTime) {
+        // TODO: impl
+    }
+
+    public var baggage: BaggageContext {
+        // TODO: impl
+        BaggageContext()
```

Comment:
> To avoid continuous back and forth copying of data between “metadata containers” it would be optimal for specific tracer implementations to depend on Baggage and have it be the storage for span identifiers and such
@ktoso

Other than performance this also has the potential benefit of using XRay-agnostic libraries to pass around the `BaggageContext`. For that to work, we'd have to have `baggage` be a stored property on `Segment`. Would that be okay with you? The actual `Instrument` on the other hand could still be implemented in a separate library so that the main library wouldn't need to depend on `Instrumentation`.

Reply:
my TODO at the moment as far as the baggage is concerned:

- define Context type, this will be equivalent of the TracingHeader but it feels weird to use "TracingHeader" in Segment
- add attribute of Context type to subsegment (will contain traceId, optionally parent and samplng decision which are separate attributes at the moment; the sampling decision is related with isRecording defined in Span)
- the Context type should eventually be the BaggageContext though it may be in 2nd iteration of changes

Reply:
> the Context type should eventually be the BaggageContext though it may be in 2nd iteration of changes

Yeah that should be the goal IMO; let us know if you'd like us to tag the baggage, we could also rename its package to skip the gsoc- already I think